src/open_dread_rando/specific_patches/mass_delete_actors.py

```diff
@@ -0,0 +1,66 @@
+from collections import namedtuple
+
+from open_dread_rando.patcher_editor import PatcherEditor
+
+ActorReferenceTuple = namedtuple("ActorReferenceTuple", ["scenario", "actor_layer", "sublayer", "actor"])
+
+def _remove_all_actors(editor: PatcherEditor, scenario_name: str, actor_layer: str,
+                       to_remove: set[ActorReferenceTuple]) -> None:
+    scenario = editor.get_scenario(scenario_name)
+
+    for sublayer_name, actor_name, actor in scenario.all_actors_in_actor_layer(actor_layer):
+        to_remove.add(ActorReferenceTuple(scenario_name, actor_layer, sublayer_name, actor_name))
+
+def _remove_actors_from_groups(editor: PatcherEditor, scenario_name: str, actor_layer: str,
+                               actor_groups: list[str], to_remove: set) -> None:
+    scenario = editor.get_scenario(scenario_name)
+
+    for group in actor_groups:
+        for actor_link in scenario.get_actor_group(group, actor_layer):
+            to_remove.add(ActorReferenceTuple(**editor.reference_for_link(actor_link, scenario_name)))
+
+def _remove_actors_not_in_groups(editor: PatcherEditor, scenario_name: str, actor_layer: str,
+                                 actor_groups: list[str], to_remove: set, to_keep) -> None:
+    scenario = editor.get_scenario(scenario_name)
+
+    for group in scenario.actor_groups_for_actor_layer(actor_layer):
+        if group not in actor_groups:
+            for actor_link in scenario.get_actor_group(group, actor_layer):
+                to_remove.add(ActorReferenceTuple(**editor.reference_for_link(actor_link, scenario_name)))
+        else:
+            for actor_link in scenario.get_actor_group(group, actor_layer):
+                to_keep.add(ActorReferenceTuple(**editor.reference_for_link(actor_link, scenario_name)))
+
+def mass_delete_actors(editor: PatcherEditor, configuration: dict) -> None:
+    # Sets of tuples will be used to ensure no duplicate entries
+    to_remove = set()
+    to_keep = set()
+
+    for reference in configuration["to_keep"]:
+        to_keep.add(ActorReferenceTuple(
+            reference["scenario"],
```

Comment:
Converting this kind of dict into an `ActorReferenceTuple` should be a separate method.

You can then even change the entire thing to something like this:
```python
to_keep = {ActorReferenceTuple.from_dict(reference) for reference in configuration["to_keep"]}
```

Reply:
(or `to_keep  = set(map(ActorReferenceTuple.from_dict , configuration["to_keep"]))` but I don't think that's better)